client/state/posts/selectors.js

```diff
@@ -90,11 +97,17 @@ export const getSitePost = createSelector(
  * @return {?Array}         Posts for the post query
  */
 export function getSitePostsForQuery( state, siteId, query ) {
-	if ( ! state.posts.queries[ siteId ] ) {
+	const manager = state.posts.queries[ siteId ];
+	if ( ! manager ) {
+		return null;
+	}
+
+	const posts = manager.getItems( query );
+	if ( ! posts ) {
 		return null;
 	}
 
-	return state.posts.queries[ siteId ].getItems( query );
+	return posts.map( ( post ) => getNormalizedPost( state, post.global_ID ) );
```

Comment:
Oh ok, thanks for the detailed explanation. I've missed the fact that the posts were duplicated in the state. So I tend to agree with maping over `getNormilizedPost` here to have only one single source of truth about posts.

But about the duplication issue, I can think also about a  (d) solution : keeping the full posts in state.post.items and storing only the ids in the QueryPostManager. This could be done using a higher order reducer. something like :

```
const postReducer = combineReducers( {
	counts,
	items,
	siteRequests,
	queryRequests,
	edits
} );

const higherOrderReducer = (state, action) => {
	const newState = postsReducer(state, action);

	return {
		...newState,
		queries: queriesEnhancedReducer(state.queries, action, newState.items)
	}
}

export default higherOrderReducer;
```

and the `queriesEnhancedReducer` will take the updated posts that can be used by the QueryManager.receive to update its posts global ids.